### PR TITLE
Tic tac toe game

### PR DESCRIPTION
**Pull request by FelipeEnne**
This is the first draft of the project.

We created a Tic Tac Toe using javascript, HTML, and CSS.

Factory functions were used to create most of the variables, like the board and the players' names.

We managed to make only a few variables global, as required in the Odin Project.

The HTML consists of a button, that shows the model that appears to the players put their names, the board that is rendered according to the status of the game and a message that tells who's the next movement and if the game ended.

The players click on the board to do a movement.

We also created a very poor AI to play against the player, but it only chooses aa random blanc space.

### CODE REVIEW

**Comment by bellom** [comment]
> This is the first draft of the project.
> 
> The tic-tac-toe it was implemented.
> 
> Kindly review the project. We will be waiting for your response.

You should add the implementation you did in the project 🙂 
Example: I added feature to input players' names.
Kindly edit 👍.

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "sourceType": "module"
+  },
+  "extends": ["airbnb-base"],
+  "rules": {
+    "no-shadow": "off",
+    "no-param-reassign": "off",
+    "eol-last": "off",
+    "arrow-parens": "off",
+    "no-unused-vars": "off",
+    "no-alert": "off"
```
Comment (bellom) [inline .eslintrc.json]: You made some changes in this file, check out the [microverse-linter-config](https://github.com/microverseinc/linters-config/tree/master/javascript)
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/

File: eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (bellom) [inline eslintrc.json]: kindly delete this file, it seems you're not using right?

File: js/index.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
+let playTurn = 1;
```
Comment (bellom) [inline js/index.js]: we shouldn't make use of global code, let use modules or factories.
image
https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript

File: js/index.js [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+let countSpaces = 0;
+let AI;
+
+
+const gameBoard = (space, value) => {
```
Comment (bellom) [inline js/index.js]: Note: The main goal of this project is for us to practice Factory Functions and The Module Pattern from the [previous section](https://www.theodinproject.com/courses/javascript/lessons/factory-functions-and-the-module-pattern) in the curriculum.
we should have `game`, `gameBoard`, and `player` objects
image

File: js/index.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+  const play = player(name);
+  play.victor();
+  return (play);
+}
+
+function updateSpace(board, space, value) {
+  board.changeSpace(space, value);
+}
+
+function render(board) {
```
Comment (bellom) [inline js/index.js]: Let separate the DOM from application logic, you can create another file for DOM.

image

https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59623090/

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @ :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**_1. I am unable to play the game, click event Listener seems not working.(nothing happens after clicking the cells to play the game)
2. Kindly check out my comments below._**

**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by mcrd25 — APPROVED** [review]
# Changes Approved 💪 🎊 🤓 

Hey Team! 👋 

Great job implementing the changes! 👏 👍 

#### suggested changes [OPTIONAL]
- The correct player's turn when playing vs AI does not update accurately. Address this [OPTIONAL]

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

**Pull request merged**